client/task-list/tasks/products/product-template-modal.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, Modal, RadioControl } from '@wordpress/components';
+import { Icon } from '@wordpress/icons';
+import { useState } from '@wordpress/element';
+import { getAdminLink } from '@woocommerce/wc-admin-settings';
+
+/**
+ * Internal dependencies
+ */
+import './product-template-modal.scss';
+import selectedRadioButtonIcon from './selected-radio-button-icon';
+
+const templates = [
+	{
+		key: 'physical_product',
+		title: __( 'Physical product', 'woocommerce-admin' ),
+	},
+	{
+		key: 'digital_product',
+		title: __( 'Digital product', 'woocommerce-admin' ),
+	},
+	{
+		key: 'variable_product',
+		title: __( 'Variable product', 'woocommerce-admin' ),
+	},
+];
+
+export function ProductTemplateModal( { onClose } ) {
+	const [ selectedTemplate, setSelectedTemplate ] = useState();
+	const [ isRedirecting, setIsRedirecting ] = useState( false );
+
+	const onSelectTemplateClick = ( template ) => {
+		setIsRedirecting( true );
+		if ( template ) {
+			window.location = getAdminLink(
+				'post-new.php?post_type=product&wc_onboarding_active_task=products&tutorial=true&product_template=' +
+					template
+			);
+		}
+		if ( onClose ) {
+			onClose();
+		}
+	};
+
+	const templateList = templates.map( ( task ) => {
+		task.onClick = () => {
+			setSelectedTemplate( task.key );
+		};
+		task.before = (
+			<div className="woocommerce-task__icon">
+				{ task.key === selectedTemplate ? (
+					<Icon icon={ selectedRadioButtonIcon } />
```

Comment:
Why was this needed? It seems the [`RadioControl`](https://developer.wordpress.org/block-editor/components/radio-control/#do) has a selected indication already?

Reply:
Thanks, didn't expect the detailed look, appreciate it :) This whole map is actually old code, and not using it.